pkg/generator/ingress_translator.go

```diff
@@ -103,14 +106,26 @@ func (translator *IngressTranslator) translateIngress(ctx context.Context, ingre
 			Namespace: ingressTLS.SecretNamespace,
 			Name:      ingressTLS.SecretName,
 		}
-		sniMatches = append(sniMatches, &envoy.SNIMatch{
+		sniMatch := &envoy.SNIMatch{
 			Hosts:            ingressTLS.Hosts,
 			CertSource:       secretRef,
-			CertificateChain: secret.Data[certFieldInSecret],
-			PrivateKey:       secret.Data[keyFieldInSecret]})
+			CertificateChain: secret.Data[certificates.CertName],
+			PrivateKey:       secret.Data[certificates.PrivateKeyName],
+		}
+
+		switch ingressTLS.Visibility {
+		// Also handle empty string as external for reverse-compatibility
```

Comment:
nit:s/reverse/backwards

Reply:
I am wondering if we need to verify ingressTLS.Visibility (previously we didn't in this controller)? 
If not this could be just: 
```
if ingressTLS.Visibility == v1alpha1.IngressVisibilityClusterLocal {
  localSNIMatches = append(localSNIMatches, sniMatch)
} else {
  externalSNIMatches = append(externalSNIMatches, sniMatch)
}
```

Reply:
I think that would also work, but could have some bad runtime behaviour if we ever add another option.